tests/test_mem_eff_attention.py

```diff
@@ -3,34 +3,120 @@
 # This source code is licensed under the BSD license found in the
 # LICENSE file in the root directory of this source tree.
 
+import math
+
 import pytest
 import torch
+from scipy.stats import binom_test
 
 import xformers.ops
 
+torch.backends.cuda.matmul.allow_tf32 = False
+cuda_only = pytest.mark.skipif(not torch.cuda.is_available(), reason="requires CUDA")
 _devices = ["cpu", "cuda"] if torch.cuda.is_available() else ["cpu"]
 
 
-def ref_attention(q, k, v):
+def assert_allclose(
```

Comment:
nit, but could be moved in some utils ? it feels like this could be used in a few places already, beyond this PR